### ISSUE TITLE
Better telemetry around TreesLatest token fetch in ODSP driver

### ISSUE THREAD

**Issue by christiango** [microsoft/FluidFramework#3894]
In our telemetry, we are leveraging two important telemetry markers. We have a top level "loadContainer" marker, that is measured by the host application. This captures the end to end time to get a container from the runtime. This includes a lot of steps along the way, such as a snapshot fetch and a storage token fetch in the ODSP case. We rely on the ODSP driver's telemetry to get insight into token fetches on the critical path for trees/latest.

One thing we have noticed is that we have way more token fetch calls logged than load container calls. Here are some examples from the past week where we had 6800 container loads, but 9000 TeesLatest_GetToken_end markers. 

Here is the query we use to get the storage token data, including a clause where we filter out the summarizer
Office_Fluid_FluidRuntime_Performance 
| where Data_documentEverHidden == false and Data_eventName == "fluid:telemetry:OdspDriver:TreesLatest_GetToken_end" and Data_clientType == "interactive"
| extend t = Data_duration 

One thing I am not sure about is if there is some retry logic going on here for expired tokens? Is this potentially getting misrepresented in telemetry? I found this code https://github.com/microsoft/FluidFramework/blob/16f58fc65c8b6693a04a75e2b9554656a3a47f5c/packages/drivers/odsp-driver/src/odspUtils.ts#L46

TL;DR:
We as consumers of fluid want a reliable signal for the amount of time spent getting a token. I'm not sure the TreesLatest_GetToken_end marker is giving us this signal, given that we are seeing way more than the expected number of calls.

**Comment by christiango** [comment]
FYI @cosmincatrinescu @ram-ramasubramanian @AndreiZe @vladsud

**Comment by christiango** [comment]
Is it possible that the sessions with more than 1 token fetch is a result of getting an expired token on the first call, then triggering a new call? If that's the case, I think we need better telemetry here so we can make this clearer to hosts. In particular, I think we'd want to know the "total blocking token time" including cases where we need to ask for another token due to an expiry (seems like we'd need more metadata in this marker).

**Comment by curtisman** [comment]
@christiangoI did some analysis of the data for the last day. 
only 1.8% of the container loaded has multiple TreesLatest_GetToken_end.

I examined some of those container load events, all the duplicate are actual are the same event.  Somehow, the event got duplicated in the database somehow (the tell is that Event_Sequence # is the same for the multipleTreesLatest_GetToken_end.  i don't think this is the runtime that emit the event multiple time.

**Comment by christiango** [comment]
@curtisman - we have reached out to the otel team where the duplicates of the same event being uploaded. It seems to be an issue with their retry logic, but in cases where this does happen, we have seen that the container load logs are also duplicated. 

So what doesn't seem to make sense here is why we have 30% more token fetch calls than container load calls (an office-web-host data point) in our data.

**Comment by curtisman** [comment]
Beyond the data we look at, there are cases where these event will emitted multiple time, for token refresh.
We will add the token refresh options to the perf event to help ease reasoning over the data.

**Comment by vladsud** [comment]
For completeness-  there are two loops right now around TreesLatest - these two events tell if any had retry:

TreeLatest_FallBackToGetRequest
TreeLatest_SecondCall

@jatgarg - what's the timeline to remove POST -> GET fallback logic?

**Comment by curtisman** [comment]
The change should be in the up coming version 0.28

**Comment by jatgarg** [comment]
> For completeness- there are two loops right now around TreesLatest - these two events tell if any had retry:
> 
> TreeLatest_FallBackToGetRequest
> TreeLatest_SecondCall
> 
> @jatgarg - what's the timeline to remove POST -> GET fallback logic?

The fallback logic will be tested in first time in 0.26. If the options call work fine without any/some fallbacks, we will remove it from main.
